examples/gno.land/p/demo/ownable/ownable.gno

```diff
@@ -4,20 +4,33 @@ import (
 	"std"
 )
 
-// Ownable is meant to be used as a top-level object to make your contract ownable OR
-// being embedded in a Gno object to manage per-object ownership.
-type Ownable struct {
+// Ownable defines an interface that requires a method to get the owner's address
+type Ownable interface {
+	Owner() std.Address
+}
+
+// Transferrable embeds Ownable, indicating that an object can transfer ownership
+type Transferrable interface {
+	Ownable
+
+	TransferOwnership(newOwner std.Address) error
+}
+
+// DefaultOwnable provides a basic implementation of ownership functionality
+// It can be used as a top-level object to make a contract ownable or embedded within
+// another object to manage per-object ownership.
+type DefaultOwnable struct {
```

Comment:
```suggestion
type TransferrableOwnership struct {
	Ownable
	Transferrable
```

Avoid using the name `DefaultOwnable`. Instead, choose a more specific name that reflects the implementation.

Embed `Ownable` and `Transferrable`, or better yet, add a check to validate that the struct implements these two interfaces.

Reply:
it makes sense